layer_analyzer.go

```diff
@@ -0,0 +1,124 @@
+package lifecycle
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+
+	"github.com/buildpacks/lifecycle/buildpack"
+	"github.com/buildpacks/lifecycle/cmd"
+	"github.com/buildpacks/lifecycle/launch"
+	"github.com/buildpacks/lifecycle/platform"
+)
+
+type LayerAnalyzer interface {
```

Comment:
Should we move the interface to the package where it's used?

Reply:
See https://github.com/buildpacks/lifecycle/pull/582#discussion_r614945193